File: projects/mi-program/003-add/php/add.php
#!/usr/bin/env php
<?php
echo "Enter a number: ";
$text = trim(fgets(STDIN));
if (!ctype_digit(ltrim($text, '-'))) {
    echo "Error: not a valid number.\n";
    exit(1);
}
$n = (int)$text;
echo "$n + 1 = " . ($n + 1) . "\n";

File: projects/mi-program/003-add/c/add.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

int main(void) {
    char input[100];
    char *end;

    printf("Enter a number: ");
    fgets(input, sizeof(input), stdin);

    long n = strtol(input, &end, 10);
    if (end == input || *end != '\n') {
        printf("Error: not a valid number.\n");
        return 1;
    }

    printf("%ld + 1 = %ld\n", n, n + 1);
    return 0;
}
